docs/ObserveOdbInteraction.md

```diff
@@ -0,0 +1,236 @@
+# Observe / ODB Interaction
+
+This document briefly sketches the information flow and ODB API calls required by Observe to execute an observation according to the latest proposal.
+
+## Fetching the Sequence(s)
+
+Observe uses an observation's `execution` field to retrieve the static (unchanging) configuration along with acquisition and science sequences.  In general, Observe will query for the complete configuration of each step and will have to be prepared to accept any instrument.  In the example below though, a small subset of the complete configuration information available (and needed) for a single instrument is shown.
+
+
+```
+fragment GmosSouthAtomFields on GmosSouthAtom {
+  steps {
+    instrumentConfig {
+      exposure { seconds }
+      filter
+      fpu {
+        builtin
+      }
+    }
+    
+    stepConfig {
+      ... on Gcal {
+        arcs
+      }
+      ... on Science {
+        offset {
+          p { arcseconds }
+          q { arcseconds }
+        }
+      }
+    }
+  }
+}
+query Observe {
+  observation(observationId: "o-2") {
+    execution {
+      executionConfig {
+        ... on GmosSouthExecutionConfig {
+          static {
+            stageMode
+            detector
+            mosPreImaging
+          }
+          acquisition {
+            nextAtom {
+              ...GmosSouthAtomFields
+            }
+            possibleFuture {
+              ...GmosSouthAtomFields
+            }
+          }
+          science {
+            nextAtom {
+              ...GmosSouthAtomFields
+            }
+            possibleFuture {
+              ...GmosSouthAtomFields
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+Notice that the `static` configuration is separated from the steps themselves and that the `acquisition` and `science` steps are segregated.  Observe would start by executing the acquisition steps and once accomplished switch to the science steps.  Within a sequence, the `nextAtom` (if any) is the one that Observe should execute while the `possibleFuture` shows what we expect to happen for the remainder of the sequence.  Because the possible future is uncertain, after finishing the `nextAtom`, Observe would repeat the query to obtain the latest `nextAtom` and latest `possibleFuture`.  This would continue until no `nextAtom` is forthcoming or until the observer decides to stop the observation.
+
+## Starting a Visit
+
+Before executing an observation, Observe would record the start of a "visit".  This produces a `VisitRecord` in the database.  In theory the `static` configuration in the query above can be modified or ignored altogether so part of recording a visit is to pass the actual static configuration that is used.
+
```

Comment:
Would a change in the observation configuration trigger a new "visit"? I think that would make sense, Observe can record a new "visit" to acknowledge the configuration change for the remaining steps.

Reply:
Yes, if you change the static configuration a new visit is needed.